src/main/java/bio/terra/service/resourcemanagement/google/GoogleResourceDao.java

```diff
@@ -27,16 +28,16 @@
 @Repository
 public class GoogleResourceDao {
     private final NamedParameterJdbcTemplate jdbcTemplate;
-    private final Connection connection;
+    private final DataSource dataSource;
 
     @Autowired
     public GoogleResourceDao(DataRepoJdbcConfiguration jdbcConfiguration) throws SQLException {
+        dataSource = jdbcConfiguration.getDataSource();
```

Comment:
This is another spot where we are doing it wrong. This constructor should be autowiring the jdbcTemplate bean. Then it doesn't need to touch the datasource or the jdbcConfiguration. The createProject method should be marked with @Transaction(REQUIRED)